Lib/json/tool.py

```diff
@@ -68,12 +72,42 @@ def main():
             outfile = open(options.outfile, 'w', encoding='utf-8')
         with outfile:
             for obj in objs:
-                json.dump(obj, outfile, **dump_args)
+                if with_colors:
+                    json_str = json.dumps(obj, **dump_args)
+                    outfile.write(colorize_json(json_str))
+                else:
+                    json.dump(obj, outfile, **dump_args)
                 outfile.write('\n')
     except ValueError as e:
         raise SystemExit(e)
 
 
+color_pattern = re.compile(r'''
+    (?P<string>"(\\.|[^"\\])*?")           |   # String
```

Comment:
I don't know if this was already addressed but:

```py
>>> import re
>>> p = re.compile(r'(\\.|[^"\\])*?')
>>> p.match('"abc"')
<re.Match object; span=(0, 0), match=''>
```

This seems to match empty strings so I'm not really sure about this.

Reply:
By the way, `json.dumps()` could output objects with `\uXXXX`: 

```py
>>> json.dumps(['\u1234'])
'["\\u1234"]'
```

Reply:
> I don't know if this was already addressed but:
> 
> ```python
> >>> import re
> >>> p = re.compile(r'(\\.|[^"\\])*?')
> >>> p.match('"abc"')
> <re.Match object; span=(0, 0), match=''>
> ```
> 
> This seems to match empty strings so I'm not really sure about this. But at the same time it works.. so I don't know.

I think you are missing the quotes in the regex here

Reply:
Oh right!

Reply:
> By the way, `json.dumps()` could output objects with `\uXXXX`:
> 
> ```python
> >>> json.dumps(['\u1234'])
> '["\\u1234"]'
> ```

I believe this is already covered, but I can make it more explicit. What's your preference?

Reply:
JSON specs expect `\u` to be followed by exactly 4 hex digits so I don't know if you want to make it stricter here (see the state machine: https://www.json.org/json-en.html)

Reply:
I can definitely reproduce the syntax exactly as given in the spec, though the regex will be longer and harder to review.

I exploited the fact that the json is known to be valid so I don't need to worry about being strict (also for the number regex).

Though I don't feel that strongly about it. If you prefer, I'm happy to write a regex that follows the spec :)

Reply:
No need. If the assumption is that the JSON is valid, it's fine (I'm also happy that you added an explicit test for that case).